### PR TITLE
GitHub rsync escaping

### PR DESCRIPTION
**Pull request by coreydaley**
Correctly escapes oc rsync command arguments

### CODE REVIEW

**Comment by bparees** [comment]
(fyi corey was testing something for me, not trying to ram his own PRs through :) )

**Comment by coreydaley** [comment]
@bparees @jim-minter added unit test

**Comment by bparees** [comment]
/lgtm

will let @jim-minter have the last word though.

**Comment by stevekuznetsov** [comment]
@bparees you may want to /lgtm cancel

**Comment by bparees** [comment]
@stevekuznetsov removing my comment and deleting the labels manually seems to have been sufficient to reign in the bot.

File: pkg/oc/cli/cmd/rsync/copyrsync_test.go [hunk 1]
```diff
@@ -22,3 +22,34 @@ func TestRshExcludeFlags(t *testing.T) {
 		}
 	})
 }
+
+// TestRsyncEscapeCommand ensures that command line options supplied to oc rsync
+// are properly escaped.
+func TestRsyncEscapeCommand(t *testing.T) {
```
Comment (jim-minter) [inline pkg/oc/cli/cmd/rsync/copyrsync_test.go]: nit: using raw strings (quote with `) will make this easier to validate

File: pkg/oc/cli/cmd/rsync/copyrsync.go [hunk 2]
```diff
@@ -111,6 +110,21 @@ func (r *rsyncStrategy) Validate() error {
 	return nil
 }
 
+// rsyncEscapeCommand wraps each element of the command in double quotes
+// if it contains any of the following: single quote, double quote, backslash, or space
+// It also replaces every pre-existing double quote character in the element with a pair.
+// Example: " -> ""
+func rsyncEscapeCommand(command []string) []string {
+	for key := range command {
+		needsQuoted := strings.ContainsAny(command[key], `'" \`)
```
Comment (jim-minter) [inline pkg/oc/cli/cmd/rsync/copyrsync.go]: don't believe the backslash is necessary in this line

File: pkg/oc/cli/cmd/rsync/copyrsync.go [hunk 3]
```diff
@@ -111,6 +110,21 @@ func (r *rsyncStrategy) Validate() error {
 	return nil
 }
 
+// rsyncEscapeCommand wraps each element of the command in double quotes
+// if it contains any of the following: single quote, double quote, backslash, or space
+// It also replaces every pre-existing double quote character in the element with a pair.
+// Example: " -> ""
+func rsyncEscapeCommand(command []string) []string {
```
Comment (jim-minter) [inline pkg/oc/cli/cmd/rsync/copyrsync.go]: note that this function modifies its argument in place and returns it, which is a little unusual

**Comment by coreydaley** [comment]
@jim-minter comments addressed

**Comment by coreydaley** [comment]
@bparees @jim-minter ptal 
comments have been addressed and tests are green

**Comment by bparees** [comment]
/approve

Ben Parees | OpenShift

**Pull request merged**